### PR TITLE
problem solved, tests passing, readme updated

### PR DESCRIPTION
**Pull request by LeoKuhorev**

### CODE REVIEW

**Comment by LeoKuhorev** [comment]
- [x] Top-level README “Table of Contents” is updated
 - [x] Feature tasks for this challenge are completed
 - [x] Unit tests written and passing
     - [x] “Happy Path” - Expected outcome
     - [ ] Expected failure
     - [x] Edge Case (if applicable/obvious)
 - [x] README for this challenge is complete
     - [x] Summary, Description, Approach & Efficiency, Solution
     - [x] Link to code
     - [x] Picture of whiteboard

File: data_structures/tree/tree.py [hunk 1]
```diff
@@ -128,6 +128,27 @@ def traverse(node):
...
+            _max = self.root.val
+
+            def traverse(node):
+                nonlocal _max
+                if node:
+                    if node.val > _max:
+                        _max = node.val
+                    traverse(node.left)
+                    traverse(node.right)
+                    return _max
```
Comment (LeoKuhorev) [inline data_structures/tree/tree.py:147]: Don't need to return, _max is modified in place
